code/datums/uplink_items/uplink_general.dm

```diff
@@ -22,25 +20,49 @@ GLOBAL_LIST_INIT(uplink_items, subtypesof(/datum/uplink_item))
 				uplink_items[I.category] = list()
 
 			uplink_items[I.category] += I
-			if(I.limited_stock < 0 && I.can_discount && I.item && I.cost > 5)
+
+			//Add items to discount pool, checking job, species, and hijacker status
+			var/job_check = FALSE
+			var/species_check = FALSE
+			var/hijacker_check = FALSE
+			if(I.job) //If your job does not match, no discount
+				for(var/job in I.job)
+					if(job == user.mind.assigned_role)
+						job_check = TRUE
+			else
+				job_check = TRUE
+			if(I.species) //If your species does not match, no discount
+				for(var/species in I.species)
+					if(species == user.dna.species.name)
+						species_check = TRUE
```

Comment:
```suggestion
				if(user.dna?.species in I.species)
					species_check = TRUE
```
Please check if this works!
If it doesn't, try using `locate(user.dna?.species)`